gridcapa-task-manager-app/src/main/java/com/farao_community/farao/gridcapa/task_manager/app/TaskDtoBuilder.java

```diff
@@ -42,50 +41,70 @@ public TaskDto getTaskDto(OffsetDateTime timestamp) {
     }
 
     public List<TaskDto> getListTasksDto(LocalDate businessDate) {
-        List<TaskDto> listTasks = new ArrayList<>();
         ZoneId zone = ZoneId.of(properties.getProcess().getTimezone());
         LocalDateTime businessDateStartTime = businessDate.atTime(0, 30);
+        LocalDateTime businessDateEndTime = businessDate.atTime(23, 30);
         ZoneOffset zoneOffSet = zone.getRules().getOffset(businessDateStartTime);
-        OffsetDateTime timestamp = businessDateStartTime.atOffset(zoneOffSet);
-        while (timestamp.getDayOfMonth() == businessDate.getDayOfMonth()) {
-            listTasks.add(getTaskDto(timestamp));
-            timestamp = timestamp.plusHours(1).atZoneSameInstant(zone).toOffsetDateTime();
+        OffsetDateTime startTimestamp = businessDateStartTime.atOffset(zoneOffSet);
+        OffsetDateTime endTimestamp = businessDateEndTime.atOffset(zoneOffSet);
+        Set<Task> tasks = taskRepository.findAllByTimestampBetweenForBusinessDayView(startTimestamp, endTimestamp);
+        Map<OffsetDateTime, TaskDto> taskMap = new HashMap<>();
+        for (OffsetDateTime loopTimestamp = startTimestamp;
+             !loopTimestamp.isAfter(endTimestamp);
+             loopTimestamp = loopTimestamp.plusHours(1).atZoneSameInstant(zone).toOffsetDateTime()
+        ) {
+            taskMap.put(loopTimestamp, getEmptyTask(loopTimestamp));
         }
-        return listTasks;
+        tasks.stream().map(t -> createDtoFromEntityWithOrWithoutEvents(t, false)).forEach(dto -> taskMap.put(dto.getTimestamp(), dto));
+
+        return taskMap.values().stream().toList();
     }
 
     public List<TaskDto> getListRunningTasksDto() {
-        return taskRepository.findAllRunningAndPending().stream().map(this::createDtoFromEntity).collect(Collectors.toUnmodifiableList());
+        return taskRepository.findAllRunningAndPending().stream().map(t -> createDtoFromEntityWithOrWithoutEvents(t, false)).toList();
     }
 
     public TaskDto getEmptyTask(OffsetDateTime timestamp) {
         return TaskDto.emptyTask(timestamp, properties.getProcess().getInputs(), properties.getProcess().getOutputs());
     }
 
     public TaskDto createDtoFromEntity(Task task) {
+        return createDtoFromEntityWithOrWithoutEvents(task, true);
+    }
+
+    private TaskDto createDtoFromEntityWithOrWithoutEvents(Task task, boolean withEvents) {
         var inputs = properties.getProcess().getInputs().stream()
-            .map(input -> task.getInput(input)
-                .map(this::createDtoFromEntity)
-                .orElseGet(() -> ProcessFileDto.emptyProcessFile(input)))
-            .collect(Collectors.toList());
+                .map(input -> task.getInput(input)
+                        .map(this::createDtoFromEntity)
+                        .orElseGet(() -> ProcessFileDto.emptyProcessFile(input)))
+                .collect(Collectors.toList());
```

Comment:
For consistency in the whole code of the class, you should use the `toList()` method instead of `collect(Collectors.toList())`.